SPECS/qemu/qemu.spec

```diff
@@ -1,64 +1,257 @@
 %endif
+%ifarch %{mips}
+%global kvm_package   system-mips
+%endif
+%ifarch riscv64
+%global kvm_package   system-riscv
+%endif
+
 %global modprobe_kvm_conf %{_sourcedir}/kvm.conf
-%ifarch x86_64
+%ifarch s390x
+    %global modprobe_kvm_conf %{_sourcedir}/kvm-s390x.conf
+%endif
+%ifarch %{ix86} x86_64
     %global modprobe_kvm_conf %{_sourcedir}/kvm-x86.conf
 %endif
+
 %global tools_only 0
+
+
+%global user_static 1
+%if 0%{?azl}
+%global user_static 0
+%endif
+%if 0%{?rhel}
+# EPEL/RHEL do not have required -static builddeps
 %global user_static 0
+%endif
+
+%global have_kvm 0
+%if 0%{?kvm_package:1}
 %global have_kvm 1
+%endif
+
+# Matches numactl ExcludeArch
 %global have_numactl 1
+%ifarch %{arm}
+%global have_numactl 0
+%endif
+
+%global have_spice 1
+%if 0%{?azl}
+%global have_spice 0
+%endif
+# Matches spice ExclusiveArch
+%ifnarch %{ix86} x86_64 %{arm} aarch64
 %global have_spice 0
+%endif
+%if 0%{?rhel} >= 9
+%global have_spice 0
+%endif
+
+# Matches xen ExclusiveArch
 %global have_xen 0
+%if 0%{?fedora}
+%ifarch x86_64 aarch64
+%global have_xen 1
+%endif
+%endif
+
+%global have_liburing 0
+%if 0%{?azl}
+%global have_liburing 1
+%endif
+%if 0%{?fedora}
+%ifnarch %{arm}
 %global have_liburing 1
+%endif
+%endif
+
+# 2.0 has this set
+%global have_virgl 0
+%if 0%{?fedora}
 %global have_virgl 1
+%endif
+
+%global have_pmem 0
+%ifarch x86_64 %{power64}
+%global have_pmem 1
+%endif
+
+%global have_jack 1
+%if 0%{?azl}
+%global have_jack 0
+%endif
+%if 0%{?rhel}
 %global have_jack 0
+%endif
+
+%global have_dbus_display 1
+%if 0%{?azl}
+%global have_dbus_display 0
+%endif
+%if %{defined rhel} && 0%{?rhel} < 9
+# RHEL/Centos 8 glib is not new enough
+%global have_dbus_display 0
+%endif
+
+%global have_libblkio 0
+%if 0%{?fedora} >= 37
+%global have_libblkio 1
+%endif
+
+%global have_gvnc_devel %{defined fedora}
+%if 0%{?azl}
+%global have_gvnc_devel 0
+%endif
+%global have_sdl_image %{defined fedora}
+%if 0%{?azl}
 %global have_sdl_image 0
+%endif
 %global have_fdt 1
 %global have_opengl 1
+%if 0%{?azl_no_ui}
+%global have_opengl 0
+%endif
 %global have_usbredir 1
 %global enable_werror 0
+
+
+# Matches edk2.spec ExclusiveArch
 %global have_edk2 0
-%global have_pmem 0
-%ifarch x86_64
-%global have_pmem 1
+%ifarch %{ix86} x86_64 %{arm} aarch64
+%global have_edk2 1
 %endif
+%if 0%{?azl}
+%global have_edk2 0
+%endif
+
 # All modules should be listed here.
 %define have_block_rbd 1
+%ifarch %{ix86} %{arm}
+%define have_block_rbd 0
+%endif
+# disable for azl till dependency is available
+%if 0%{?azl}
+%define have_block_rbd 0
+%endif
+
+
+%global have_block_gluster 1
+%if 0%{?azl}
 %global have_block_gluster 0
+%endif
+%if 0%{?rhel} >= 9
+%global have_block_gluster 0
+%endif
+
+%define have_block_nfs 0
+%if 0%{?azl}
 %define have_block_nfs 1
+%endif
+%if 0%{?fedora}
+%define have_block_nfs 1
+%endif
+
 %define have_librdma 1
+%ifarch %{arm}
+%define have_librdma 0
+%endif
+
 %define have_libcacard 1
+%if 0%{?azl}
+%define have_libcacard 0
```

Comment:
2.0 set have_libacard  to 1, why 0 now?

Reply:
This adds support for 'emulation of smart cards to a virtual card reader running in a guest virtual machine'. Do we really need it? I have intentionally turned of some subpackages that I consider not useful to us in the near future. We can easily turn on the support when required.

Reply:
i have no idea if we need it, but removing it is a breaking change we should make purposefully.  i suspect we are fine to remove, but i'm not sure who knows best here.

Reply:
I believe BMP is the only real user of qemu on mariner so far. We should be able to make a call on it.

Reply:
i think anyone can create a mariner VM in azure, right?  there is a marketplace image.  if we make a breaking change to qemu, i think we should consider more than just BMP scenario.

Reply:
talked to britel, you are totally right.  if we get subsequent requests to reenable, we can revisit.